logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -11,13 +12,15 @@ import com.wire.kalium.network.api.user.client.ClientApi
 import com.wire.kalium.network.utils.isSuccessful
 import com.wire.kalium.persistence.dao.ConversationDAO
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.map
 
 interface ConversationRepository {
     suspend fun fetchConversations(): Either<CoreFailure, Unit>
     suspend fun getConversationList(): Flow<List<Conversation>>
     suspend fun getConversationDetails(conversationId: ConversationId): Either<CoreFailure, Conversation>
+    suspend fun getConversationDetailsFromDB(conversationId: QualifiedID): Flow<Conversation>
```

Comment:
```suggestion
    suspend fun observeConversationDetails(conversationId: ConversationId): Flow<Conversation>
```
To whoever is consuming the Repository, It shouldn't matter where it comes from. Also, it's better to use `ConversationId`, in case it ever changes in the future. Like it recently did (from single string to qualified).


On a side note, I think `getConversationDetails` shouldn't be the way it is. I don't think we will fetch things from remote sources all the time as we rely heavily on sync and backend events to keep data up-to-date.
Maybe the refactor spirits will give it a go soon when nobody is expecting it 🙏🏼